Mailbox.Tests/ProgramTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MailRoom;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MailRoom.Tests
+{
+    [TestClass]
+    public class ProgramTests
+    {
+        [TestMethod]
+        public void GetOwnersDisplay_ConcatenatesUniqueOwners_ReturnsString()
+        {
+            List<Mailbox> boxes = new List<Mailbox>
+            {
+                new Mailbox(new Person("F1", "L1"), (0, 0), Sizes.Small),
+                new Mailbox(new Person("F1", "L1"), (1, 0), Sizes.Small),
+                new Mailbox(new Person("F2", "L2"), (2, 0), Sizes.Small)
+            };
+            string expected = boxes[0].Owner.ToString() + Environment.NewLine + boxes[2].Owner.ToString();
+            MailboxCollection mailboxes = new MailboxCollection(boxes, 3, 1);
+
+            string result = Program.GetOwnersDisplay(mailboxes);
+
+            Assert.AreEqual<string>(expected, result);
+        }
+
+        [DataTestMethod]
+        public void GetOwnersDisplay_GivenNullCollection_ReturnsEmptyString()
+        {
+#pragma warning disable CS8625 // testing in case null collection does get passed somehow
+            string? result = Program.GetOwnersDisplay(null);
+#pragma warning restore CS8625 // Cannot convert null literal to non-nullable reference type.
+
+            Assert.AreEqual<string>("", result);
+        }
+
+        [DataTestMethod]
+        public void GetMailboxDetails_GetsStringRepresentationOfMailbox_ReturnsConcatenatedMailbox()
+        {
+            Mailbox box = new Mailbox(new Person("F1", "L1"), (0, 0), Sizes.Small);
+            MailboxCollection mailboxes = new MailboxCollection(new List<Mailbox> { box }, 1, 1);
+            string expected = box.ToString();
+
+            string result = Program.GetMailboxDetails(mailboxes, 0, 0) ?? "";
+
+            Assert.AreEqual<string>(expected, result);

```

Comment:
In general you don't need to specify the generic parameter type unless the values passed in are of different types. The compiler can infer the correct generic type.
```suggestion
            Assert.AreEqual(expected, result);
```